### DISCUSSION TITLE
How to do app rename on update?

### DISCUSSION THREAD

**Discussion by maniac-coder** [sparkle-project/Sparkle#2261] [Q&A] [answered]
I went through this https://github.com/sparkle-project/Sparkle/pull/1444 and followed the steps mentioned in it. But it doesn't seem to work for me. I am not sure what I am missing. Can someone elaborate on how to achieve this? 

I am using Sparkle 1.27.0 via cocoapods.

**Comment by zorgiepoo** [comment, marked as answer]
You need to compile your own build of Sparkle setting `SPARKLE_NORMALIZE_INSTALLED_APPLICATION_NAME = 1` in `ConfigCommon.xcconfig`, then set `SUBundleName` or `CFBundleName` in your app, deliver an update, and then on the next update after that the rename will take effect. And maybe you want to switch appcasts, but that's up to you. And finally, disable and set `SPARKLE_NORMALIZE_INSTALLED_APPLICATION_NAME = 0` when you're done and it's no longer needed.

The normalization process isn't that great. Users can generally rename an app bundle to whatever they want, and this setting forces a specific rename which may override the user's choice. That's why it's not supported as easily as you may like.

**Reply by maniac-coder** [reply]
Thanks for the detailed explanation.

Making two updates to have the name changes seems lengthy and more than 80% of the user won't change the name. Kindly consider making this simple.
